crates/noirc_evaluator/src/brillig/brillig_ir.rs

```diff
@@ -31,15 +27,33 @@ use acvm::{
 /// constrained functions in terms of syntax compatibility.
 pub(crate) const BRILLIG_INTEGER_ARITHMETIC_BIT_SIZE: u32 = 127;
 
+// Registers reserved in runtime for special purposes.
+pub(crate) enum ReservedRegisters {
+    /// This register stores the free memory pointer. Allocations must be done after this pointer.
+    Alloc = 0,
```

Comment:
I currently think of Alloc = StackPointer by analogy with https://stackoverflow.com/questions/1395591/what-is-exactly-the-base-pointer-and-stack-pointer-to-what-do-they-point

and https://man7.org/linux/man-pages/man3/alloca.3.html which uses the stack pointer